### PR TITLE
RPP_Agents_Java_JUnit_LastErrorLog

### PR DESCRIPTION
**Pull request by entrudo**
Add Last Error Log to RP

### CODE REVIEW

File: src/main/java/com/epam/reportportal/junit/ReportPortalListener.java [hunk 1]
```diff
@@ -320,7 +324,14 @@ protected void stopRunner(@Nonnull final Object runner) {
 			l.setStatus(status);
 			if (l.getParentId() == null) {
 				rq.setStatus(ofNullable(status).map(Enum::name).orElse(null));
+				if (!failedTestsMessages.isEmpty()) {
```
Comment (HardNorth) [inline src/main/java/com/epam/reportportal/junit/ReportPortalListener.java:327]: You don't need to add errors on Suites

File: src/main/java/com/epam/reportportal/junit/ReportPortalListener.java [hunk 2]
```diff
@@ -582,6 +593,12 @@ protected void stopTestMethod(@Nonnull final Object runner, @Nonnull final Frame
 	protected void stopTestMethod(@Nonnull final Object runner, @Nonnull final FrameworkMethod method,
 			@Nonnull final ReflectiveCallable callable, @Nonnull final ItemStatus status, @Nullable final Throwable throwable) {
 		FinishTestItemRQ rq = buildFinishStepRq(runner, method, callable, status);
+		if (status != ItemStatus.PASSED && throwable != null) {
+			rq.setDescription(String.format(DESCRIPTION_TEST_ERROR_FORMAT,
```
Comment (HardNorth) [inline src/main/java/com/epam/reportportal/junit/ReportPortalListener.java:597]: Please don't set anything into requests outside "build" methods.

File: src/main/java/com/epam/reportportal/junit/ReportPortalListener.java [hunk 3]
```diff
@@ -582,6 +593,12 @@ protected void stopTestMethod(@Nonnull final Object runner, @Nonnull final Frame
...
+					createStepDescription(this.context.getTestMethodDescription(method), method),
+					throwable.getClass().getSimpleName(),
+					ofNullable(throwable.getMessage()).isPresent() ? throwable.getMessage() : ""));
```
Comment (HardNorth) [inline src/main/java/com/epam/reportportal/junit/ReportPortalListener.java:600]: Las error log contains StackTrace, not just error message. Please put it into the description. Please use `ExceptionUtils.removeCommonFrames` method to simplify Stack Trace, but putting just message is not informative.

**Review by HardNorth — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/reportportal/junit/ReportPortalListener.java (hunk 1 above)
Comment (entrudo) [reply src/main/java/com/epam/reportportal/junit/ReportPortalListener.java]: Updated, I removed it from Suites level, it was discussed with Andrey.

File: src/main/java/com/epam/reportportal/junit/ReportPortalListener.java (hunk 2 above)
Comment (entrudo) [reply src/main/java/com/epam/reportportal/junit/ReportPortalListener.java]: Moved it to the build method.

File: src/main/java/com/epam/reportportal/junit/ReportPortalListener.java (hunk 3 above)
Comment (entrudo) [reply src/main/java/com/epam/reportportal/junit/ReportPortalListener.java]: This type of message I confirmed with Andrey. I changed it to the full StackTrace.

**Pull request merged**